src/bin/flux-lsp.rs

```diff
@@ -31,14 +49,67 @@ async fn main() {
         .unwrap();
     }
 
-    log::debug!("Starting lsp client");
-    let stdin = tokio::io::stdin();
-    let stdout = tokio::io::stdout();
-
     let (service, messages) =
         LspService::new(|client| LspServer::new(Some(client)));
-    Server::new(stdin, stdout)
-        .interleave(messages)
-        .serve(service)
-        .await;
+
+    let channel =
+        matches.channel.unwrap_or_else(|| "stdio".to_string());
+    match channel.as_str() {
+        "stdio" => {
+            log::debug!("Communicating using stdin/stdout");
+            Server::new(tokio::io::stdin(), tokio::io::stdout())
+                .interleave(messages)
+                .serve(service)
+                .await;
```

Comment:
Creates a server using stdin and stdout for its communication channel (currently the only way to use the lsp), loops in this `messages` channel to the communication chain, and serves until `SIGINT`. I cannot, for the life of me, remember what `messages` is for, but I feel like it might have something to do with internally generated messages.